plugins/maciVoting/hooks/useProposal.tsx

```diff
@@ -119,6 +115,8 @@ export function useProposal(proposalId: string, autoRefresh = false) {
 
   return {
     proposal,
+    proposalMetadata: metadataContent,
+    creator: proposalCreationEvent?.creator,
```

Comment:
When fetching a specific proposal we could avoid the `getLogs` loop in here https://github.com/privacy-scaling-explorations/maci-aragon-osx-gov-app/blob/06e13042f67654cbba2aa0e18906384e30e12a9d/plugins/maciVoting/hooks/useProposal.tsx#L61

if we save the metadata (and the creator address inside it) in the plugin as mappings
```javascript
metadata(proposalId) = {
  creator: "0x123...",
  // ...
}
```

This could be done in a future PR

Reply:
I see. Do you think we should do this? Since Aragon has moved away from storing metadata onchain. We don't have to ofc but imagine reason would have been minimising onchain storage costs

It looks like the metadata is actually just used to [compute the proposalId](https://github.com/aragon/token-voting-plugin/blob/b20983c1a69b82a14b243b6415cc165b6b223858/packages/contracts/src/TokenVoting.sol#L144C71-L144C80) in the token voting plugin

Reply:
It is true 🤔: minimizing RPC calls vs minimising onchain storage costs. I would save in onchain costs